linode/lke/cluster.go

```diff
@@ -489,3 +489,40 @@ func expandLKEClusterControlPlane(controlPlane map[string]interface{}) linodego.
 
 	return result
 }
+
+func removeExternalPools(ctx context.Context, externalPoolTags []string, pools []linodego.LKENodePool) []linodego.LKENodePool {
+	var filteredPools []linodego.LKENodePool
+
+	if len(externalPoolTags) == 0 {
+		return pools
+	}
+
+	tagSet := make(map[string]struct{}, len(externalPoolTags))
+	for _, tag := range externalPoolTags {
+		tagSet[tag] = struct{}{}
+	}
+
+	for _, pool := range pools {
+		tag := poolHasAnyOfTags(pool, tagSet)
+		if tag == "" {
+			filteredPools = append(filteredPools, pool)
+		} else {
+			tflog.Info(ctx, "Excluding pool from management by this resource", map[string]interface{}{
+				"pool_id": pool.ID,
+				"tag":     tag,
+				"reason":  "Pool tagged to be managed by a separate linode_lke_node_pool resource",
+			})
+		}
+	}
+
+	return filteredPools
+}
```

Comment:
Optional stylistic suggestions, feel free to ignore 🙂 

```suggestion
func filterExternalPools(ctx context.Context, externalPoolTags []string, pools []linodego.LKENodePool) []linodego.LKENodePool {
	var filteredPools []linodego.LKENodePool

	if len(externalPoolTags) == 0 {
		return pools
	}

	tagSet := make(map[string]bool, len(externalPoolTags))
	for _, tag := range externalPoolTags {
		tagSet[tag] = true
	}

	for _, pool := range pools {
		tag := poolHasAnyOfTags(pool, tagSet)

		if tag != "" {
			tflog.Info(ctx, "Excluding pool from management by this resource", map[string]interface{}{
				"pool_id": pool.ID,
				"tag":     tag,
				"reason":  "Pool tagged to be managed by a separate linode_lke_node_pool resource",
			})
			continue
		}

		filteredPools = append(filteredPools, pool)
	}

	return filteredPools
}
```

Reply:
done